app/utils/commands.js

```diff
@@ -0,0 +1,253 @@
+const { UntilMorning } = require('./untilMorning')
+const log = require('electron-log')
+
+const allOptions = {
+  title: {
+    long: '--title',
+    short: '-T',
+    description: 'Specify title for next break (Long or Mini)',
+    withValue: true
+  },
+  text: {
+    long: '--text',
+    short: '-t',
+    description: 'Specify text for next break (Long Break only)',
+    withValue: true
+  },
+  skip: {
+    long: '--skip',
+    short: '-s',
+    description: 'Skip directly to this break (Long or Mini)',
+    withValue: false
+  },
+  duration: {
+    long: '--duration',
+    short: '-d',
+    description: 'Specify duration for pausing breaks (Pause only) [infinitely|until-morning|seconds (as number)]',
+    withValue: true
+  }
+}
+
+const allCommands = {
+  help: {
+    description: 'Show this help message'
+  },
+  version: {
+    description: 'Show current stretchly version'
+  },
+  reset: {
+    description: 'Reset breaks'
+  },
+  pause: {
+    description: 'Pause breaks',
+    options: [allOptions.duration]
+  },
+  resume: {
+    description: 'Resume from a pause'
+  },
+  toggle: {
+    description: 'Toggle breaks between resume/paused'
+  },
+  mini: {
+    description: 'Skips to next Mini Break',
+    options: [allOptions.title, allOptions.skip]
+  },
+  long: {
+    description: 'Skips to next Long Break',
+    options: [allOptions.text, allOptions.title, allOptions.skip]
+  }
+}
+
+const allExamples = [{
+  cmd: 'stretchly pause',
+  description: 'Pause breaks indefinitly'
+},
+{
+  cmd: 'stretchly pause -d 3600',
+  description: 'Pause breaks for one hour'
+},
+{
+  cmd: 'stretchly mini -T "Stretch up !" --skip',
+  description: 'Skips to next Mini Break with "Stretch up!" title'
+},
+{
+  cmd: 'stretchly long -T "Stretch up !" -t "Go stretch !"',
+  description: 'Set next long break title to "Stretch up !" and text to "Go stretch !"'
+}]
+
+// Parse cmd line, check if valid and put variables in a dedicated object
+class Command {
+  constructor (input, version) {
+    this.version = version
+    this.supported = allCommands
+
+    this.parse(input)
+  }
+
+  parse (input) {
+    // filter out electron flags first
+    let i = 0
+    while (input[i].startsWith('--')) {
```

Comment:
`npm run dev` gives error

```
TypeError: Cannot read property 'startsWith' of undefined
```

Reply:
Ok, it triggers when you have an already running instance, fixing it

Reply:
Fixed by https://github.com/hovancik/stretchly/pull/700/commits/1424058ee0283fa2dbab94784456718c93f8908a